cmd/argocd/commands/admin/app.go

```diff
@@ -437,7 +442,7 @@ func reconcileApplications(
 		sources = append(sources, app.Spec.GetSource())
 		revisions = append(revisions, app.Spec.GetSource().TargetRevision)
 
-		res, err := appStateManager.CompareAppState(&app, proj, revisions, sources, false, false, nil, false, false)
+		res, err := appStateManager.CompareAppState(destCluster, &app, proj, revisions, sources, false, false, nil, false, false)
```

Comment:
I feel like it would be better to do the inference again in CompareAppState instead of adding a parameter here.

Reply:
As discussed, this is a refactor and can be improved further in another PR. No blocker.